### ISSUE TITLE
processCitations does not find CSL file

### ISSUE THREAD

**Issue by monofon** [jgm/pandoc#7055]
I tried using this this code

```haskell
main = do
  md <- Text.readFile "document.md"
  html <-
    handleError $
      runPure $
        setResourcePath ["."]
          >> readMarkdown def md
          >>= processCitations
          >>= writeHtml5String def {writerCiteMethod = Citeproc}
  Text.writeFile "document.html" html
```

to convert a small Markdown fragment to HTML and process the citations on the way.

```markdown
---
title: test
csl: acm-sig-proceedings
bibliography: bibliography.bib
---

@10.1145/3406088.3409014

# Bibliography
```

The layout of the directory is

```
├── Main.hs
├── bibliography.bib
├── acm-sig-proceedings.csl
└── document.md
```

My assumption was that `processCitations` would pick up ` acm-sig-proceedings.csl` right away, like the old plugin did, but it doesn't. The resulting error message is

```
Could not find data file data/data/default.csl
```

I looked into `Text.Pandoc.Citeproc.getStyle` but could not see a reason for this happening.

However, running Pandoc directly works and the provided CSL file is found and used.

```sh
pandoc -C  document.md -o document.html
```

So, I assume I am doing something wrong here, but this may also be a bug.

Running pandoc-2.11.4 on macOS.

**Comment by jgm** [comment]
You're using `runPure` which cannot do IO.  It uses an ersatz file system implemented in CommonState.  If you really need this to be pure, you need to populate this with the files beforehand. Otherwise just use `runIO` and it should work.

**Comment by monofon** [comment]
Ah yes, that was it. Thanks a lot.

**Comment by kaushalmodi** [comment]
@jgm I believe I am running into the same issue, but I am using the last released static binary of `pandoc`, version 2.11.4. I noticed that the `csl` metadata specified doesn't get used at all, and the `@` references stay as-is after pandoc processing.
